File: app/src/main/java/com/arcao/opencaching4locus/ui/base/util/PermissionUtil.java
package com.arcao.opencaching4locus.ui.base.util;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.support.annotation.NonNull;
import android.support.v4.content.ContextCompat;

public class PermissionUtil {
	private static final int REQUEST_PERMISSION_BASE = 100;
	public static final int REQUEST_LOCATION_PERMISSION = REQUEST_PERMISSION_BASE + 1;
	public static final int REQUEST_EXTERNAL_STORAGE_PERMISSION = REQUEST_PERMISSION_BASE + 2;

	public static final String[] PERMISSION_LOCATION_GPS =
			{ Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION};
	public static final String[] PERMISSION_LOCATION_WIFI =
			{ Manifest.permission.ACCESS_COARSE_LOCATION};
	public static final String[] PERMISSION_EXTERNAL_STORAGE =
			{ Manifest.permission.WRITE_EXTERNAL_STORAGE};

	public static boolean verifyPermissions(@NonNull int[] grantResults) {
		for (int result : grantResults) {
			if (result != PackageManager.PERMISSION_GRANTED) {
				return false;
			}
		}
		return true;
	}

	public static boolean hasPermission(Context context, String... permissions) {
		for (String permission : permissions) {
			if (ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_DENIED)
				return false;
		}

		return true;
	}
}

File: app/src/main/kotlin/com/arcao/opencaching4locus/ui/base/util/PermissionUtil.kt
package com.arcao.opencaching4locus.ui.base.util

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import android.support.v4.content.ContextCompat

object PermissionUtil {
    private val REQUEST_PERMISSION_BASE = 100
    val REQUEST_LOCATION_PERMISSION = REQUEST_PERMISSION_BASE + 1
    val REQUEST_EXTERNAL_STORAGE_PERMISSION = REQUEST_PERMISSION_BASE + 2

    val PERMISSION_LOCATION_GPS = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION)
    val PERMISSION_LOCATION_WIFI = arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION)
    val PERMISSION_EXTERNAL_STORAGE = arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE)
}

fun Context.verifyPermissions(grantResults: IntArray): Boolean {
    return grantResults.none {
        it != PackageManager.PERMISSION_GRANTED
    }
}

fun Context.hasPermission(vararg permissions: String): Boolean {
    return permissions.none {
        ContextCompat.checkSelfPermission(this, it) == PackageManager.PERMISSION_DENIED
    }
}
